packages/slate/src/models/node.js

```diff
@@ -1371,41 +1389,54 @@ class Node {
    */
 
   getParent(key) {
-    if (this.hasChild(key)) return this
+    if (!this.hasDescendant(key)) return null
 
-    let node = null
-
-    this.nodes.find(child => {
-      if (child.object == 'text') {
-        return false
-      } else {
-        node = child.getParent(key)
-        return node
-      }
-    })
-
-    return node
+    const path = this.getPath(key)
+    const parentPath = path.slice(0, path.length - 1)
+    return this.getDescendantAtPath(parentPath)
   }
 
   /**
    * Get the path of a descendant node by `key`.
    *
    * @param {String|Node} key
-   * @return {Array}
+   * @return {Array|null}
    */
 
   getPath(key) {
-    let child = this.assertNode(key)
-    const ancestors = this.getAncestors(key)
-    const path = []
+    key = assertKey(key)
+    if (this.key === key) return []
+    if (this.nodes.size === 0) return null
+    let result = null
+    // PERF: often the node keys is generated with sequences;
```

Comment:
This is not really the kind of performance improvement I'd like to add here, since it's so implementation-specific. People can swap out their key generator with whatever they want. I'd need to see this in an isolated pull request to see if it's really quick enough to merit adding.

Reply:
Fixed~